### PR TITLE
Suggested fix for #1173 with suitable tests

### PR DESCRIPTION
**Pull request by cgnuechtel**
Fixes issue #1173 and adds tests for expected step results

### CODE REVIEW

File: Functions/Gherkin.Tests.ps1 [hunk 1]
```diff
@@ -258,11 +258,57 @@ Describe "When displaying PesterResults in the console" -Tag Gherkin {
...
+                return 'Inconclusive'
+            }
+            default { return 'Passed' }
+        }
+    }
+
+    for ($i = 0; $i -lt $gherkin.Results.TestResult.Count; $i++) {
+        $expectedResult = (Get-ExpectedResult $i)
+        $result = $gherkin.Results.TestResult[$i]
+        It "Test result $($i + 1) ('$($result.Name)') should be '$expectedResult'" {
```
[the remark this answers is not in the document]
Comment (nohwnd) [reply Functions/Gherkin.Tests.ps1]: I think we forgot to add test exclusions to the appveyor build. The style/help/whitespace checks should run only when I try releasing. Can’t check right now.

[the remark this answers is not in the document]
Comment (nohwnd) [reply Functions/Gherkin.Tests.ps1]: I am reacting to the other comment :/

[the remark this answers is not in the document]
Comment (Jaykul) [reply Functions/Gherkin.Tests.ps1]: What?

File: Functions/Assertions/Set-TestInconclusive.ps1 [hunk 2]
```diff
@@ -1,3 +1,13 @@
+function New-InconclusiveErrorRecord ([string] $Message, [string] $File, [string] $Line, [string] $LineText) {
```
[the remark this answers is not in the document]
Comment (cgnuechtel) [reply Functions/Assertions/Set-TestInconclusive.ps1]: I just restored the original code.
Does anybody know more about this code?
@michalporeba?

File: Functions/Gherkin.Tests.ps1 (hunk 1 above)
[the remark this answers is not in the document]
Comment (cgnuechtel) [reply Functions/Gherkin.Tests.ps1]: > I really dislike code that generates tests -- it's extremely hard to tell what these tests prove, if anything...

You are right.
That code is a little bit too dynamic.
The test logic must be simpler.
I will replace the function by simple hash tables later...

My goal was to create a data-driven test

[the remark this answers is not in the document]
Comment (cgnuechtel) [reply Functions/Gherkin.Tests.ps1]: @Jaykul I changed test code and think now it is more readable

File: Functions/Assertions/Set-TestInconclusive.ps1 (hunk 2 above)
[the remark this answers is not in the document]
Comment (Jaykul) [reply Functions/Assertions/Set-TestInconclusive.ps1]: I guess I don't know what you mean. Is this function being used anywhere? Did this file have two functions in it earlier, or did that one get renamed and refactored to `Set-TestInconclusive`?

Regardless, I don't think it makes sense _in this file_, nor do I think we should have any code written in this PowerShell 1.0 style, with the parameters outside the block and no comment help 😉

[the remark this answers is not in the document]
Comment (nohwnd) [reply Functions/Assertions/Set-TestInconclusive.ps1]: `New-InconclusiveErrorRecord` was replaced by `Set-ItResult`, you probably solved merge conflicts incorrectly at some point.

**Comment by nohwnd** [comment]
@cgnuechtel I upgraded your branch to the latest master because I changed how builds are done. It failed on PowerShell 2 because PowerShell 2 does not automatically expand properties on arrays. Calling `.Result` on an array fails because array does not have a property called Result (unlike in PowerShell 3+, where it looks for .Result on all items in the array). (There is now fail on using PSJob after it was disposed, it happens occasionally, I will be fixing that in a different branch and update this to current master when I am done.)

**Comment by cgnuechtel** [comment]
@nohwnd Thanks, but Set-ItResult does not yet have a `File` parameter!?

**Comment by nohwnd** [comment]
@cgnuechtel I see it builds now, is this still work in progress or done?

**Comment by cgnuechtel** [comment]
@nohwnd I am ready here, but please look at Set-ItResult.ps1 as I added some parameters. However, it should be a non-breaking change.

**Comment by nohwnd** [comment]
@cgnuechtel thanks for the heads up, I think using the Set-ItResult directly was not a good idea (I know I changed your code to do that, so it's my bad.) I will refactor to make it call an internal function that creates the error. The public `Set-ItResult` should not be able to define anything except for status and reason.

File: Functions/Assertions/Set-TestInconclusive.ps1 (hunk 2 above)
[the remark this answers is not in the document]
Comment (cgnuechtel) [reply Functions/Assertions/Set-TestInconclusive.ps1]: > `New-InconclusiveErrorRecord` was replaced by `Set-ItResult`, you probably solved merge conflicts incorrectly at some point.

@nohwnd That code is in the [master](https://github.com/pester/Pester/blob/master/Functions/Gherkin.ps1#L699) and was not updated by pull request #1026 which removed `New-InconclusiveErrorRecord`.

**Comment by nohwnd** [comment]
@cgnuechtel it was the correct change to make I think, but not for the `Set-ItResult` there when the failure has a known type I want to show a custom message build from the `because` string. There are no tests for it because it's very close to the output I guess. Now it should be working as expected. 

``` 
PS C:\projects\pester_main> get-module pester | remove-module ; import-module .\Pester.psd1; describe "a"{ it "b" { set-itresult -inconclusive  }}

Describing a

Hit Line breakpoint on 'C:\projects\pester_main\Functions\Output.ps1:286'

[DBG]: PS C:\projects\pester_main>
  [?] b, is inconclusive 67ms
PS C:\projects\pester_main> get-module pester | remove-module ; import-module .\Pester.psd1; describe "a"{ it "b" { set-itresult -inconclusive -Because "abc" }}

Describing a

[DBG]: PS C:\projects\pester_main>
  [?] b, is inconclusive, because abc 73ms
```

Hope I did not break anything else.

**Comment by nohwnd** [comment]
All tests pass, merging, if it does not work we should prove it wrong with broken tests.

**Pull request merged**